mtp_cashbook/apps/disbursements/utils.py

```diff
@@ -49,3 +50,77 @@ def get_disbursement_viability(request, disbursement):
     )
 
     return viability
+
+
+def find_addresses(postcode):
+    try:
+        results = requests.get(
+            settings.POSTCODE_LOOKUP_ENDPOINT,
+            params={'postcode': postcode, 'key': settings.POSTCODE_LOOKUP_AUTH_TOKEN},
+            timeout=5
+        )
+    except RequestException:
+        return []
+
+    addresses = []
+    if results.status_code == 200:
+        address_results = results.json().get('results', [])
+        for result in address_results:
+            address_data = result['DPA']
+
+            address = {
+                'address': address_data['ADDRESS'],
+                'city': address_data['POST_TOWN'],
+                'postcode': address_data['POSTCODE'],
+            }
+
+            business = build_address_line(
+                address_data, 'DEPARTMENT_NAME', 'ORGANISATION_NAME'
+            )
+
+            po_box = address_data.get('PO_BOX_NUMBER')
+
+            building = build_address_line(
+                address_data, 'SUB_BUILDING_NAME', 'BUILDING_NAME'
+            )
+
+            building_number = address_data.get('BUILDING_NUMBER')
+            street = build_address_line(
+                address_data, 'DEPENDENT_THOROUGHFARE_NAME', 'THOROUGHFARE_NAME'
+            )
+            if building_number and street:
+                street_number = ' '.join([building_number, street])
+            else:
+                street_number = building_number or street
+
+            locality = build_address_line(
+                address_data, 'DOUBLE_DEPENDENT_LOCALITY', 'DEPENDENT_LOCALITY'
+            )
+
+            line_1 = ''
+            line_2 = ''
+
+            for item in [business, po_box, building, street_number, locality]:
```

Comment:
maybe `line1, *line2 = list(filter(None, [business, po_box, building, street_number, locality])); line2 = ', '.join(line2)` or is that going too far? it's just that we're like 6 indents deep…